packages/flutter/lib/src/material/input.dart

```diff
@@ -137,6 +147,15 @@ class _InputState extends State<Input> {
       config.onSubmitted(_value);
   }
 
+  void _handleSelectionChanged(TextSelection selection) {
+    if (_isAttachedToKeyboard) {
+      _keyboardHandle.setSelection(selection.start, selection.end);
```

Comment:
I think we should organise the keyboard handle and the editable string such that you tell the same object what the new selection is regardless of whether a keyboard is attached.